runtime-infra/pn-data-monitoring.yaml

```diff
@@ -0,0 +1,417 @@
+          - Effect: Allow
+            Action:
+              - "s3:PutObject"
+            Resource: 
+              - !Sub "arn:aws:s3:::${DataMonitoringBucket}/${ParquetBucketPath}/*"       
+          # Accessi in lettura al bucket dump dell'account confinfo nella cartella parquet
+          - Fn::If:
+            - IsCoreAccount
+            - Effect: Allow
+              Action:
+                - "s3:Get*"
+                - "s3:List*"
+              Resource: 
+              - !Sub "arn:aws:s3:::${ConfinfoDynamoDbExportBucketName}/${ParquetBucketPath}/*"       
+            - !Ref AWS::NoValue
+          # Accessi in lettura ai log di pn-delivery
+          - Fn::If:
+            - IsCoreAccount
+            - Effect: Allow
+              Action:
+                - "logs:StartQuery"
+                - "logs:StopQuery"
+                - "logs:GetQueryResults"
+              Resource: 
+              - !Sub "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/ecs/pn-delivery:*"
+            - !Ref AWS::NoValue
+          # Allow the use of encryption key
+          - Effect: Allow
+            Action:
+              - kms:Encrypt
+              - kms:Decrypt
+              - kms:ReEncrypt*
+              - kms:GenerateDataKey*
+              - kms:DescribeKey
+            Resource: 
+              - !Sub "${LogsBucketKmsKeyArn}"
+
+  DataMonitoringScheduleRole:
+    Condition: HasCronExpression
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: events.amazonaws.com
+        Version: "2012-10-17"
+      Policies:
+        - PolicyName: runCodeBuild
+          PolicyDocument:
+            Statement:
+              - Sid: startProjectRun
+                Action:
+                  - "codebuild:*"
+                Effect: Allow
+                Resource: 
+                  - !GetAtt DataMonitoringCodebuildProject.Arn
+
+  IndexScheduledRule:
+    Type: AWS::Events::Rule
+    Condition: HasCronExpression
+    Properties:
+      Description: "pn-data-monitoring-codebuild-scheduled-rule"
+      # every day at 9 am
+      ScheduleExpression: !Ref DataMonitoringIndexCronExpression 
+      State: "ENABLED"
+      RoleArn: !GetAtt "DataMonitoringScheduleRole.Arn"
+      Targets: 
+        - Id: "DataMonitoringIndex"
+          RoleArn: !GetAtt "DataMonitoringScheduleRole.Arn"
+          # Input:
+          #   !Sub |
+          #     {
+          #       "environmentVariablesOverride": {
+          #         "cmd": ${Cmd}
+          #       }
+          #     }
+          Arn: 
+            Fn::GetAtt: 
+              - "DataMonitoringCodebuildProject"
+              - "Arn"
+        
+  DynamoDbExportFunction:
+    Type: AWS::Lambda::Function
+    Condition: HasDynamoDbExportTableNames
+    Properties:
+      Runtime: python3.12
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You must specify a valid value for Runtime (python3.12). Valid values are [&#34;dotnet6&#34;, &#34;dotnetcore1.0&#34;, &#34;dotnetcore2.0&#34;, &#34;dotnetcore2.1&#34;, &#34;dotnetcore3.1&#34;, &#34;go1.x&#34;, &#34;java11&#34;, &#34;java17&#34;, &#34;java8&#34;, &#34;java8.al2&#34;, &#34;nodejs&#34;, &#34;nodejs10.x&#34;, &#34;nodejs12.x&#34;, &#34;nodejs14.x&#34;, &#34;nodejs16.x&#34;, &#34;nodejs18.x&#34;, &#34;nodejs4.3&#34;, &#34;nodejs4.3-edge&#34;, &#34;nodejs6.10&#34;, &#34;nodejs8.10&#34;, &#34;provided&#34;, &#34;provided.al2&#34;, &#34;python2.7&#34;, &#34;python3.10&#34;, &#34;python3.11&#34;, &#34;python3.6&#34;, &#34;python3.7&#34;, &#34;python3.8&#34;, &#34;python3.9&#34;, &#34;ruby2.5&#34;, &#34;ruby2.7&#34;, &#34;ruby3.2&#34;]